File: Daily Challanges/2026-04-April/2026-04-26.js
function explodeFizzbuzz(targetZCount) {
  let letters = Array.from("fizzbuzz");
  let steps = 0;
  while (letters.filter((c) => c === "z").length < targetZCount) {
    steps++;
    let newLetters = [];
    for (let i = 0; i < letters.length; i++) {
      let pos = i + 1;
      if (pos % 15 === 0) {
        newLetters.push(..."fizzbuzz");
      } else if (pos % 3 === 0) {
        newLetters.push(..."fizz");
      } else if (pos % 5 === 0) {
        newLetters.push(..."buzz");
      } else {
        newLetters.push(letters[i]);
      }
    }
    letters = newLetters;
  }
  return steps;
}

console.log(explodeFizzbuzz(9));
console.log(explodeFizzbuzz(15));
console.log(explodeFizzbuzz(51));
console.log(explodeFizzbuzz(52));

File: Daily Challanges/2026-04-April/2026-04-26.py
def explode_fizzbuzz(target_z_count):
    letters = list("fizzbuzz")
    steps = 0
    while letters.count('z') < target_z_count:
        steps += 1
        new_letters = []
        for i, letter in enumerate(letters, start=1):
            if i % 15 == 0:
                new_letters.extend("fizzbuzz")
            elif i % 3 == 0:
                new_letters.extend("fizz")
            elif i % 5 == 0:
                new_letters.extend("buzz")
            else:
                new_letters.append(letter)
        letters = new_letters
    return steps

print(explode_fizzbuzz(9))
print(explode_fizzbuzz(15))
print(explode_fizzbuzz(51))
print(explode_fizzbuzz(52))